### PR TITLE
chore(instance-status): Require password to save new instance settings

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

This solves issue #1 of [findings](https://posthog.slack.com/archives/C05276ME9T5/p1683611543861409).

| Before | After |
|--------|--------|
| before | after |

## How did you test this code?

Added some API tests.

### CODE REVIEW

File: posthog/api/instance_settings.py [hunk 1]
```diff
@@ -122,6 +123,13 @@ class InstanceSettingsViewset(
     serializer_class = InstanceSettingsSerializer
     lookup_field = "key"
 
+    def get_authenticators(self) -> List[BaseAuthentication]:
+        if self.request.method not in permissions.SAFE_METHODS:
+            # Write operations can only be performed with email + password auth
+            # This way knowledge of user credentials is REQUIRED to e.g. extract email credentials via the canary email
```
Comment (thmsobrmlr) [inline posthog/api/instance_settings.py:129]:
```suggestion
            # This way knowledge of user credentials is REQUIRED to e.g. get the email password via the email host setting
```

**Review by thmsobrmlr — APPROVED** [review]
Code change does what it says. Extra props for updating the kea logic format :)

I'm not super familiar with self-hosting though - is it possible to use SSO there as well? This would lock out staff users without email + password credentials.

**Comment by Twixes** [comment]
Ohh, that's a very good point with SSO. It would be really complex to add support for requiring SSO reauthentication… Now on second and third thought, I'm really thinking requiring extra authentication here is overkill, just because this endpoint is only available to staff users, and as staff you already have access to the Django admin interface. This solution was probably a bit rushed then, and isn't _really_ useful. I'll scale the PR down to just a refactor of instance settings loading.

**Comment by Twixes** [comment]
Closing in favor of #15645.

**Pull request closed without merging**